pentaho-platform-plugin-deployer/src/main/java/org/pentaho/osgi/platform/plugin/deployer/impl/PlatformDeployerBundleListener.java

```diff
@@ -0,0 +1,76 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Pentaho : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+package org.pentaho.osgi.platform.plugin.deployer.impl;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.BundleListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a bundle listener that gets the callback whenever the bundle state changes.
+ */
+
+public class PlatformDeployerBundleListener implements BundleListener {
+  private static Logger log = LoggerFactory.getLogger( PlatformDeployerBundleListener.class );
+  private BundleContext bundleContext;
+  private BundleStateManager bundleStateManager;
+  private static final String COMMON_UI_BUNDLE = "common-ui";
+
+  public void setBundleContext( BundleContext bundleContext ) {
+    this.bundleContext = bundleContext;
+  }
+
+  public void setBundleStateManager( BundleStateManager bundleStateManager ) {
+    this.bundleStateManager = bundleStateManager;
+  }
+
+  // For unit test only
+  static void setLog( Logger log ) {
+    PlatformDeployerBundleListener.log = log;
+  }
+
+  // For unit test only
+  static Logger getLog() {
+    return log;
+  }
+
+  @Override public void bundleChanged( BundleEvent event ) {
+    switch ( event.getType() ) {
+      case BundleEvent.UNINSTALLED:
+      case BundleEvent.INSTALLED:
+        // Checking for "common-ui" for now
+        if ( event.getBundle().getSymbolicName().startsWith( COMMON_UI_BUNDLE ) ) {
+          String bundleStr = event.getBundle().getHeaders().get( "Bundle-Name" ) +  event.getBundle().getHeaders().get( "Bundle-Version" );
```

Comment:
I did start out by comparing symbolic name. The symbolic name we receive in bundleEvent is "common-ui "(which is read from plugin.xml) whereas the PlatformPluginBundlingURLConnection which calls the isBundleInstalled() has the name as "common-ui-7.1-QAT". So we had to read the headers.

Thanks a lot for reviewing the code.